src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -82,22 +82,34 @@ def get_set_cmd_dicts(out: str) -> List[dict]:
                 if not spec:
                     logger.warning('No ServiceSpec found for %s', dd)
                     continue
-                if not all([spec.api_user, spec.api_password]):
-                    reason = 'api_user or api_password is not specified in ServiceSpec'
-                    logger.warning(
-                        'Unable to add iSCSI gateway to the Dashboard for %s: %s', dd, reason)
-                    continue
-                host = self._inventory_get_addr(dd.hostname)
-                service_url = 'http://{}:{}@{}:{}'.format(
-                    spec.api_user, spec.api_password, host, spec.api_port or '5000')
+                if not spec.api_user:
+                    spec.api_user = "admin"
+                if not spec.api_password:
+                    spec.api_password = "admin"
```

Comment:
```suggestion
```

these are already set to a default in the `IscsiServiceSpec` ?

Reply:
Hmm..not quite sure if a default credential is a good idea.

Reply:
> Hmm..not quite sure if a default credential is a good idea.

We used a default setting with Ceph Ansible. Nobody claimed about this fact.